torch/_dynamo/symbolic_convert.py

```diff
@@ -788,7 +792,15 @@ def LOAD_DEREF(self, inst):
         self.push(self.symbolic_locals[inst.argval])
 
     def STORE_FAST(self, inst):
-        self.symbolic_locals[inst.argval] = self.pop()
+        loaded_vt = self.pop()
+        name = inst.argval
+        # If the last top VT in the stack (just popped) is a handle (see [On tensor.register_hook]), we associate
+        # the stored name.
+        if isinstance(loaded_vt, RemovableHandleVariable):
+            new_name = self.output.new_var(name)
+            new_rhv = loaded_vt.clone(last_seen_name=new_name)
+            loaded_vt = self.replace_all(loaded_vt, new_rhv)
+        self.symbolic_locals[name] = loaded_vt
```

Comment:
No, just no. Figure out another way to do this.

Reply:
This is temporary, I'll hide this somewhere better (like `loaded_vt.rename(tx, name)`)